File: s1/s1/src/test/java/com/nodal/s1/publisher/RabbitMQPublisherTest.java
package com.nodal.s1.publisher;

import com.nodal.s1.consumer.RabbitMQConsumer;
import org.junit.jupiter.api.Test;
import org.springframework.amqp.rabbit.core.RabbitTemplate;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.mock.mockito.MockBean;

@SpringBootTest
public class RabbitMQPublisherTest {
    @MockBean
    private RabbitTemplate rabbitTemplate;

    @Autowired
    private RabbitMQPublisher publisher;

    @Test
    public void testsendMessage(){
        publisher.sendMessage("Ping");
    }

}

File: s2/s2/src/test/kotlin/com/nodal/s2/publisher/RabbitMQPublisherTest.kt
package com.nodal.s2.publisher

import org.junit.jupiter.api.Test
import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.mock.mockito.MockBean

@SpringBootTest
class RabbitMQPublisherTest {
    @MockBean
    private val rabbitTemplate: RabbitTemplate? = null

    @Autowired
    private lateinit var publisher: RabbitMQPublisher

    @Test
    fun testsendMessage() {
        publisher.sendMessage("Ping")
    }
}